### PR TITLE
chore(integer): make oveflowing_add/sub return BooleanBlock

### PR DESCRIPTION
**Pull request by tmontaigu**

### CODE REVIEW

File: tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs [hunk 1]
```diff
@@ -1817,15 +1816,14 @@ where
         let (ct_res, result_overflowed) = executor.execute((&ctxt_0, &ctxt_1));
         let (tmp_ct, tmp_o) = executor.execute((&ctxt_0, &ctxt_1));
         assert!(ct_res.block_carries_are_empty());
-        assert!(result_overflowed.carry_is_empty());
```
Comment (IceTDrinker) [inline tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs:1820]: in what case can the carry be non empty ?

Comment (tmontaigu) [reply tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs:1820]: Since its a boolean block, carries are always empty

Comment (IceTDrinker) [reply tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs:1820]: that's very nice, cool thanks !

**Review by IceTDrinker — APPROVED** [review]
Thanks !

**Pull request merged**